django/core/serializers/python.py

```diff
@@ -96,100 +96,120 @@ def getvalue(self):
         return self.objects
 
 
-def Deserializer(
-    object_list, *, using=DEFAULT_DB_ALIAS, ignorenonexistent=False, **options
-):
+class Deserializer:
```

Comment:
We already have `django.core.serializers.base.Deserializer` which the xml deserializer inherits
Then `json`, `jsonl` and `pyaml` each have `Deserializer` functions still, it would be great if this could be adapted so that these are all classes that inherit from `python.Deserializer` and the python `Deserializer` inherits from `base.Deserializer`
The current state is a bit messy which I think makes it harder to document :thinking:

Reply:
Fair enough. So, like what happens in the `xml_serializer`, creating the object list in `json`, `jsonl`, and `pyaml` which are now created using their `Deserializer` helper functions in `django/core/serializers/json.py`, `django/core/serializers/jsonl.py`, and `django/core/serializers/pyaml.py`, respectively, should become a method in the `Deserializer` class (the equivalent of `pulldom.parse(self.stream, self._make_parser())` in the `xml_deserializer`). For each three of them, there must be a class to implement that object creation🤔